src/storage-types/src/connections/aws.rs

```diff
@@ -94,34 +174,45 @@ impl RustType<ProtoAwsCredentials> for AwsCredentials {
     }
 }
 
-/// A role for Materialize to assume when performing AWS API calls.
+/// Details required for Materialize to assume customer's role
 #[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct AwsAssumeRole {
     /// The Amazon Resource Name of the role to assume.
     pub arn: String,
+    /// The optional session name for the assume role session.
+    pub session_name: Option<String>,
+    /// The Materialize AWS principal details.
+    pub mz_principal: MaterializePrincipal,
 }
 
 impl RustType<ProtoAwsAssumeRole> for AwsAssumeRole {
     fn into_proto(&self) -> ProtoAwsAssumeRole {
         ProtoAwsAssumeRole {
             arn: self.arn.clone(),
+            session_name: self.session_name.clone(),
+            mz_principal: Some(self.mz_principal.into_proto()),
         }
     }
 
     fn from_proto(proto: ProtoAwsAssumeRole) -> Result<Self, TryFromProtoError> {
-        Ok(AwsAssumeRole { arn: proto.arn })
+        Ok(AwsAssumeRole {
+            arn: proto.arn,
+            session_name: proto.session_name,
+            mz_principal: proto
+                .mz_principal
+                .into_rust_if_some("ProtoAwsAssumeRole::mz_principal")?,
+        })
     }
 }
 
 impl AwsConfig {
     /// Loads the AWS SDK configuration object from the environment, then
     /// applies the overrides from this object.
-    pub async fn load(
+    pub async fn get_sdk_config(
```

Comment:
```suggestion
    pub async fn load_sdk_config(
```

Recommend sticking with "load" in the name to indicate this function does work (vs being a simple accessor).